src/QMCDrivers/WFOpt/EngineHandle.h

```diff
@@ -118,14 +118,22 @@ class LMYEngineHandle : public EngineHandle
   cqmc::engine::LMYEngine<Value>& lm_engine_;
   std::vector<FullPrecValue> der_rat_samp;
   std::vector<FullPrecValue> le_der_samp;
+  std::vector <int> sample_count;
 
 public:
   LMYEngineHandle(cqmc::engine::LMYEngine<Value>& lmyEngine) : lm_engine_(lmyEngine){};
 
-  void prepareSampling(int num_params) override
+  void prepareSampling(int num_params,int num_samples) override
   {
+      int num_threads = omp_get_max_threads();
     der_rat_samp.resize(num_params + 1, 0.0);
     le_der_samp.resize(num_params + 1, 0.0);
+    if(lm_engine_.getStoringSamples())
+    {
+        sample_count.resize(num_threads,0);
```

Comment:
I'd encourage you to read the section on [thread safety of stl containers](https://en.cppreference.com/w/cpp/container)